aiocache/base.py

```diff
@@ -498,12 +498,13 @@ async def close(self, *args, _conn=None, **kwargs):
     async def _close(self, *args, **kwargs):
         pass
 
-    def _build_key(self, key, namespace=None):
-        if namespace is not None:
-            return "{}{}".format(namespace, _ensure_key(key))
-        if self.namespace is not None:
-            return "{}{}".format(self.namespace, _ensure_key(key))
-        return key
+    def build_key(self, key: str, namespace: Optional[str] = None) -> str:
+        key_name = key.value if isinstance(key, Enum) else key
+        ns = namespace if namespace is not None else self.namespace
+        return self._build_key(key_name, namespace=ns)
```

Comment:
```suggestion
        return self._build_key(key_name, ns)
```
Our typing guarantees 2 positional arguments, it doesn't guarantee the keyword name.

Reply:
Good point.